opensora/models/layers/blocks.py

```diff
@@ -499,8 +499,8 @@ def forward(self, x, cond, mask=None):
 
         # shape:
         # q, k, v: [B, SUB_N, NUM_HEADS, HEAD_DIM]
-        q = self.q_linear(x).view(1, -1, self.num_heads, self.head_dim)
-        kv = self.kv_linear(cond).view(1, -1, 2, self.num_heads, self.head_dim)
+        q = self.q_linear(x).view(B, -1, self.num_heads, self.head_dim)
+        kv = self.kv_linear(cond).view(B, -1, 2, self.num_heads, self.head_dim)
```

Comment:
The shape of `cond` is `[1, TOKEN_SUM_OF_BATCH, C]`, and token sum of batch may not be divisible by `B`. See https://github.com/hpcaitech/Open-Sora-dev/blob/8db3e09a02929652baa4835711a84fc9eb0df215/opensora/models/stdit/stdit3.py#L347

Reply:
However, I think the modification about `q` is correct.

Reply:
I see. Yes I made a mistake in assuming the shape of `cond`. I will create another PR for this issue.